server/dataone_register.py

```diff
@@ -148,7 +148,9 @@ def find_initial_pid(path):
         return re.sub(
             r'^http[s]?://cn.dataone.org/cn/d1/v[\d]/\w+/', '', path)
     elif re.search('https:\/\/cn-stage-2.test.dataone.org\/cn\/v2\/resolve\/', path):
-        return re.sub('https:\/\/cn-stage-2.test.dataone.org\/cn\/v2\/resolve\/', '', path)
+        return re.sub('https:\/\/cn-stage-2.test.dataone.org\/cn\/v2\/resolve\/', path)
+    elif re.search(r'^http[s]?:\/\/cn.dataone.org\/cn\/v2\/resolve\/', path):
```

Comment:
I bet we could make this whole section a little more DRY by just relaxing the regex or not using a regex at all. Maybe just strsplit on `/resolve/` and take the token immediately after?